packages/api-explorer/src/components/DocStatus/DocStatus.tsx

```diff
@@ -1,38 +1,80 @@
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
 import React, { FC } from 'react'
-import { Badge } from '@looker/components'
-import styled from 'styled-components'
+import { generatePressed, intentUIBlend } from '@looker/design-tokens'
+import styled, { css } from 'styled-components'
 import { IMethod } from '@looker/sdk-codegen'
 
 interface DocStatusProps {
   method: IMethod
 }
 
-const StyledBadge = styled(Badge)`
-  border-radius: 4px;
-
-  &.beta {
-    background-color: ${(props) => props.theme.colors.palette.purple100};
-    color: ${(props) => props.theme.colors.palette.purple700};
+const getStatusColor = (status: string) => {
+  switch (status) {
+    case 'beta':
+      return 'key'
+    case 'stable':
+      return 'positive'
+    case 'experimental':
+      return 'warn'
+    case 'deprecated':
+      return 'critical'
+    default:
+      return 'inform'
   }
+}
 
-  &.stable {
-    background-color: ${(props) => props.theme.colors.palette.green100};
-    color: ${(props) => props.theme.colors.palette.green700};
-  }
+type BadgeIntent =
+  | 'warn'
+  | 'positive'
+  | 'critical'
+  | 'inform'
+  | 'neutral'
+  | 'key'
 
-  &.experimental {
-    background-color: ${(props) => props.theme.colors.palette.yellow000};
-    color: ${(props) => props.theme.colors.palette.yellow600};
-  }
+const statusIntent = (intent: BadgeIntent) =>
+  css`
+    background: ${intentUIBlend(intent, 1)};
+    color: ${({ theme }) => generatePressed(theme.colors[intent])};
+  `
 
-  &.deprecated {
-    background-color: ${(props) => props.theme.colors.palette.red500};
-    color: ${(props) => props.theme.colors.palette.white};
-  }
+const StatusBadgeInternal = styled.div<DocStatusProps>`
+  ${(props) => statusIntent(getStatusColor(props.method.status))};
+  border-radius: 4px;
+  font-size: ${({ theme }) => `${theme.fontSizes.small}`};
+  font-weight: ${({ theme }) => `${theme.fontWeights.semiBold}`};
+  padding: ${({
+    theme: {
+      space: { xxsmall, xsmall },
+    },
+  }) => `${xxsmall} ${xsmall}`};
 `
 
 export const DocStatus: FC<DocStatusProps> = ({ method }) => (
-  <StyledBadge className={method.status.toLowerCase()}>
-    {method.status.toUpperCase()}
-  </StyledBadge>
+  <StatusBadgeInternal method={method} className={method.status}>
+    {method.status.toLocaleUpperCase()}
+  </StatusBadgeInternal>
 )
```

Comment:
This is very similar to `MethodBadge` and can probably be refactored in a similar fashion. In fact the two are so similar that I think we could combine them.